### PR TITLE
PAE - 159 setup stubs organisation registration accreditation

### PR DESCRIPTION
**Pull request by mohammeddebashi**
Ticket: [PAE-159](https://eaflood.atlassian.net/browse/PAE-159)
## Description

created stubs for signup, registration, accreditation

---

Please see the [Pull Requests standards](https://defra.github.io/software-development-standards/processes/pull_requests).

[PAE-159]: https://eaflood.atlassian.net/browse/PAE-159?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/common/enums/event.js [hunk 1]
```diff
@@ -4,7 +4,8 @@ export const LOGGING_EVENT_CATEGORIES = {
   HTTP: 'http',
   PROXY: 'proxy',
   SECRET: 'secret',
-  SERVER: 'server'
+  SERVER: 'server',
```
[the remark this answers is not in the document]
Comment (mohammeddebashi) [reply src/common/enums/event.js:7]: I guess server is related to server as it is starting or failing, but API would be related to methods relating to our api endpoints

File: postman/epr-backend.postman_collection.json [hunk 2]
```diff
@@ -67,6 +67,72 @@
...
+          "mode": "raw",
+          "raw": "{\n  \"meta\": {\n    \"schemaVersion\": \"2\",\n    \"timestamp\": \"2025-03-25T10:00:00Z\"\n  },\n  \"data\": {\n    \"main\": {\n      \"componentName\": \"Some value from the form\",\n      \"richComponentName\": { \"foo\": \"bar\", \"baz\": true }\n    },\n    \"repeaters\": {\n      \"repeaterName\": [\n        { \"textComponentName\": \"First repeater value\" },\n        { \"richComponentName\": { \"foo\": \"bar\", \"baz\": true } }\n      ]\n    },\n    \"files\": {\n      \"fileComponentName\": [\n        {\n          \"fileId\": \"123-456-789\",\n          \"link\": \"https://forms-designer/file-download/123-456-789\"\n        }\n      ]\n    }\n  }\n}"
+        },
+        "url": {
+          "raw": "{{host}}/v1/apply/accreditation",
+          "host": ["{{host}}"],
+          "path": ["send-email"]
+        }
+      },
+      "response": []
```
Comment (nijk) [inline postman/epr-backend.postman_collection.json:135]: We need to remove the old endpoints that have been removed

File: fixtures/organisation.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          "email": "mark.lee@greencycle.co.uk"
+        }
+      ]
+    },
+    "files": {
+      "supportingDocuments": [
+        {
+          "fileId": "org-987-654",
+          "link": "https://forms-designer/file-download/org-987-654"
+        }
```
Comment (nijk) [inline fixtures/organisation.json:43]: No need for files in this test, there are none in the real form as far as I know

File: src/routes/v1/apply/registration.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { createLogger } from '../../../common/helpers/logging/logger.js'
+import {
+  LOGGING_EVENT_ACTIONS,
+  LOGGING_EVENT_CATEGORIES
+} from '../../../common/enums/event.js'
+import Joi from 'joi'
```
Comment (nijk) [inline src/routes/v1/apply/registration.js:6]: I thought we were removing Joi

File: postman/epr-backend.postman_collection.json [hunk 5]
```diff
@@ -67,6 +67,72 @@
         }
       },
       "response": []
+    },
```
Comment (nijk) [inline postman/epr-backend.postman_collection.json:70]: All three of these endpoints fail with:

```
{
    "statusCode": 400,
    "error": "Bad Request",
    "message": "Invalid payload",
    "validation": {
        "source": "payload",
        "keys": []
    }
}
```

Comment (nijk) [reply postman/epr-backend.postman_collection.json:70]: yes, the problem is that the `path` values are incorrect

File: src/routes/v1/apply/organisation.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      payload: request.payload
+    })
+
+    return h
+      .response({
+        orgId: 'ORG12345', // TODO: generate correctly structured orgId to be specified and done in later ticket
+        orgName: 'ORGABCD', // depending on the field names from the form creators, given in the response
+        referenceNumber: 'REF12345' // TODO: generate correctly structured reference number to be specified and done in later ticket
+      })
+      .code(HTTP_STATUS.OK)
```
[the remark this answers is not in the document]
Comment (mohammeddebashi) [reply src/routes/v1/apply/organisation.js:45]: if we don't specify and there is no content the default would be a `204`, because the is no content in the response.
So either we ` return h.response().code(HTTP_STATUS.OK)`, ` return h.response('')` or  we accept that it will return a `204`

File: postman/epr-backend.postman_collection.json (hunk 5 above)
Comment (mohammeddebashi) [reply postman/epr-backend.postman_collection.json:44]: im testing them they are fine for me

File: postman/epr-backend.postman_collection.json [hunk 7]
```diff
@@ -21,33 +21,51 @@
...
+        "body": {
+          "mode": "raw",
+          "raw": "{\n  \"meta\": {\n    \"schemaVersion\": \"2\",\n    \"timestamp\": \"2025-03-25T10:00:00Z\"\n  },\n  \"data\": {\n    \"main\": {\n      \"componentName\": \"Some value from the form\",\n      \"richComponentName\": { \"foo\": \"bar\", \"baz\": true }\n    },\n    \"repeaters\": {\n      \"repeaterName\": [\n        { \"textComponentName\": \"First repeater value\" },\n        { \"richComponentName\": { \"foo\": \"bar\", \"baz\": true } }\n      ]\n    },\n    \"files\": {\n      \"fileComponentName\": [\n        {\n          \"fileId\": \"123-456-789\",\n          \"link\": \"https://forms-designer/file-download/123-456-789\"\n        }\n      ]\n    }\n  }\n}"
+        },
         "url": {
-          "raw": "{{host}}/example/1",
+          "raw": "{{host}}/v1/apply/organisation",
           "host": ["{{host}}"],
-          "path": ["example", "1"]
+          "path": ["send-email"]
```
Comment (nijk) [inline postman/epr-backend.postman_collection.json:62]:
```suggestion
          "path": ["v1", "apply", "organisation"]
```

File: postman/epr-backend.postman_collection.json (hunk 5 above)
Comment (nijk) [reply postman/epr-backend.postman_collection.json:44]: Code fixes suggested below

**Review by nijk — DISMISSED** [review]
